src/boundaries/stagger_and_step.jl

```diff
@@ -0,0 +1,217 @@
+export stagger_and_step!, stagger_trajectory!
+using LinearAlgebra:norm
+using Random
+using ProgressMeter
+
+
+"""
+    stagger_trajectory!(ds, x0, Tm, isinside; kwargs...) -> xi
+
+This function returns a point `xi` which _guarantees_ `T(xi) > 
+Tm` with a random walk search around the initial coordinates 
+`x0`. `T(xi)` is the escape time of the initial condition `xi`.  
+There is no guarantee to find such point, the parameters 
+must be adjusted to find the suitable point. This is an .
+auxiliary function for [`stagger_and_step`](@ref). Keyword arguments and 
+definitions are identical for both functions. 
+
+The initial search radius `δ₀` is big, `δ₀ = 1.` by default.
+
+"""
+function stagger_trajectory!(ds, x0, Tm, isinside; δ₀ = 1., stagger_mode = :exp, max_steps = Int(1e5), f = 1.1)
+    T = escape_time!(ds, x0, isinside)
+    xi = deepcopy(x0) 
+    while !(T > Tm)  # we must have T > Tm at each step 
+        xi, T = get_stagger!(ds, xi, δ₀, T, isinside; f, stagger_mode, max_steps)
+        if T < 0
+            error("Cannot find a stagger trajectory. Choose a different starting point or search radius δ₀.")
+        end 
+            
+    end
+    return xi
+end
+
+
+"""
+    stagger_and_step!(ds::DynamicalSystem, x0, N::Int, isinside::Function; kwargs...) -> trajectory
+
+This function implements the stagger-and-step method 
+[^Sweet2001] and [^Sala2016] to approximate the invariant 
```

Comment:
please change citations from `[^Sweet]` to use our BiBTeX integration with `@cite` like in the rest of the library.